### PR TITLE
RN-870 Migrate validation service to GHA

### PR DESCRIPTION
**Pull request by biaoli0**
### Issue #:
RN-870

### Changes:
1. Upgrade yarn from 3.2.1 to 3.2.4, as GHA is not compatible with v.3.2.1 for module caching 
2. Migrate validation service to GHA 

CI: 
https://github.com/beyondessential/tupaia/actions/runs/4738148278

### CODE REVIEW

File: .github/workflows/validation.yaml [hunk 1]
```diff
@@ -0,0 +1,52 @@
+name: Validation
+on:
+  push:
+    branches:
+      - '*'
+  workflow_dispatch:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref || github.run_id }}
+  cancel-in-progress: true
```
Comment (rohan-bes) [inline .github/workflows/validation.yaml:10]: This cancels the running job if a new one comes up?

File: .github/workflows/validation.yaml [hunk 2]
```diff
@@ -0,0 +1,52 @@
+name: Validation
+on:
+  push:
+    branches:
+      - '*'
+  workflow_dispatch:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref || github.run_id }}
```
Comment (rohan-bes) [inline .github/workflows/validation.yaml:9]: When is the `ref` undefined?

File: .github/workflows/validation.yaml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  validate_tests:
+    name: Validate unit tests
+    strategy:
+      matrix:
+        node:
+          - '14.19.3'
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
```
Comment (rohan-bes) [inline .github/workflows/validation.yaml:43]: Oh this sets up node? Cool 😎

**Review by rohan-bes — APPROVED** [review]
Looks great! Thanks @biaoli0 

I just had questions really about how things work, but I've enjoyed playing around with GHA!

File: .github/workflows/validation.yaml (hunk 2 above)
Comment (biaoli0) [reply .github/workflows/validation.yaml:9]: This is a very good question, it is copy and paste from Tamanu workflow and I didn't look into that deep. 

`github.ref` would be the branch name for Tupaia in most cases, unless someone uses `git tag` and leave `undefined` message, `github.ref` could be undefined here.

File: .github/workflows/validation.yaml (hunk 3 above)
Comment (biaoli0) [reply .github/workflows/validation.yaml:43]: Yeah GHA use this kind of format to call actions. You can create a new action or call a third party action from marketplace too.

File: .github/workflows/validation.yaml (hunk 1 above)
Comment (biaoli0) [reply .github/workflows/validation.yaml:10]: This will cancel the running jobs if a new commit pushed to the same branch.

**Pull request merged**